### PR TITLE
Add warning concerning the migration of customized admin credentials

### PR DESCRIPTION
**Pull request by Ebaneck**
**Component**:

'docs'

**Context**: 

See #2125 

**Summary**:

MetalK8s administrators should be aware of the breaking changes in 2.5.0 regarding 
the new authentication mechanism and how to migrate their old creds.

Should land in 2.4.3!!

**Acceptance criteria**: 

---

Closes: #2125

### CODE REVIEW

File: docs/operation/upgrade.rst [hunk 1]
```diff
@@ -24,14 +24,40 @@ Upgrade Pre-requisites
...
 
-     /srv/scality/metalk8s-X.X.X/upgrade.sh --destination-version <destination_version> --dry-run --verbose
+     /srv/scality/metalk8s-X.X.X/upgrade.sh --destination-version \
+       <destination_version> --dry-run --verbose
+
+Backup old credentials
+----------------------
+Starting 2.5.0, MetalK8s will henceforth implement OpenID Connect (OIDC) based
+authentication for both the MetalK8s UI and Grafana UI. Both UIs now make use
+of a common user store and ofcourse a common access credential.
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]: Not sure we want to talk about UIs here: it's K8s and Grafana APIs which are authenticated.
Now, an interesting note to give to users is that both K8s and Grafana will be configured to use the same OIDC provider, as you stated in the last sentence (but using "OIDC provider" instead of "user store and access credential" would make it more obvious, IMHO).

File: docs/operation/upgrade.rst [hunk 2]
```diff
@@ -24,14 +24,40 @@ Upgrade Pre-requisites
...
+
+.. warning::
+
+    To run an upgrade to version 2.5.0, MetalK8s administrators **must**
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]:
```suggestion
    Before running an upgrade from 2.4.x to 2.5.0 or higher, MetalK8s administrators **must**
```

File: docs/operation/upgrade.rst [hunk 3]
```diff
@@ -24,14 +24,40 @@ Upgrade Pre-requisites
...
+    perform a manual backup/save the passwords set for MetalK8s UI and Grafana
+    UI. Failure to save these credentials will mean administrators accept
+    losing old credentials upon upgrade.
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]: No, actually we shouldn't formulate it like this. Instead, say that the upgrade procedure will be destructive, and that passwords will be reset to their default values, so the user needs to save them.

File: docs/operation/upgrade.rst [hunk 4]
```diff
@@ -24,14 +24,40 @@ Upgrade Pre-requisites
...
+
+After upgrade is complete, administrators can later follow
+:ref:`this procedure <Change-dex-static-user-password>` to re-instate/restore
+admin credentials.
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]: Not sure referencing these docs in 2.4 will work well. Instead, maybe just mention that a procedure for configuring the OIDC provider's user store will be provided in the next version.

File: docs/operation/upgrade.rst [hunk 5]
```diff
@@ -24,14 +24,40 @@ Upgrade Pre-requisites
...
+Backup old credentials
+----------------------
+Starting 2.5.0, MetalK8s will henceforth implement OpenID Connect (OIDC) based
+authentication for both the MetalK8s UI and Grafana UI. Both UIs now make use
+of a common user store and ofcourse a common access credential.
+
+.. warning::
+
+    To run an upgrade to version 2.5.0, MetalK8s administrators **must**
+    perform a manual backup/save the passwords set for MetalK8s UI and Grafana
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]: We can't really "backup" anything, they are passwords. Once set, they are encrypted, so no way to put those back into a new system. You need to "remember" them :innocent: 

Maybe rephrase to:
```suggestion
    ensure all static users defined in `/etc/htpasswd` can be recreated,
    and if any, all users that were defined in Grafana.
```

**Review by gdemonet — CHANGES_REQUESTED** [review]
The backup note should actually be reworded more strongly IMHO (cc @thomasdanan):

We are going to **lose** all the users/passwords that were set, both in `/etc/htpasswd` and in Grafana configuration (any user created from Grafana UI would be lost on reboot today, even in 2.6, because Grafana isn't configured to use persistent storage - only the admin password can be saved in configuration).

The note should focus on this, because there will be no form of "restoration from a backup" as one could expect. It's going to be a complete re-creation of the user store (and maybe even remove some static users to replace them with an external provider, now that it's available as well :wink:).

File: docs/operation/upgrade.rst (hunk 5 above)
Comment (Ebaneck) [reply docs/operation/upgrade.rst]: Done

File: docs/operation/upgrade.rst (hunk 4 above)
Comment (Ebaneck) [reply docs/operation/upgrade.rst]: Done

File: docs/operation/upgrade.rst (hunk 3 above)
Comment (Ebaneck) [reply docs/operation/upgrade.rst]: ok

File: docs/operation/upgrade.rst (hunk 1 above)
Comment (Ebaneck) [reply docs/operation/upgrade.rst]: Done

File: docs/operation/upgrade.rst [hunk 6]
```diff
@@ -24,14 +24,43 @@ Upgrade Pre-requisites
...
    .. code::
 
-     /srv/scality/metalk8s-X.X.X/upgrade.sh --destination-version <destination_version> --dry-run --verbose
+     /srv/scality/metalk8s-X.X.X/upgrade.sh --destination-version \
+       <destination_version> --dry-run --verbose
+
+Backup old credentials
+----------------------
+Starting 2.5.0, MetalK8s will henceforth implement OpenID Connect (OIDC) based
+authentication. Both K8s and Grafana will be configured to make use of thesame
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]:
```suggestion
authentication. Both K8s and Grafana will be configured to make use of the same
```

File: docs/operation/upgrade.rst [hunk 7]
```diff
@@ -24,14 +24,43 @@ Upgrade Pre-requisites
...
+OIDC provider.
+
+.. warning::
+
+    Before running an upgrade from 2.4.x to 2.5.0 or higher, MetalK8s
+    administrators **must** ensure all static users defined in
+    ``/etc/kubernetes/htpasswd`` can be recreated, and if any, all users that
+    were defined in Grafana.
+    The upgrade procedure from 2.4.x to 2.5.0 or higher will result to admin
+    set usernames and passwords being resetted to default values. MetalK8s
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]:
```suggestion
    The upgrade procedure will result in all admin credentialss being reset
    to their default values, and any additional user being removed. MetalK8s
```

File: docs/operation/upgrade.rst [hunk 8]
```diff
@@ -24,14 +24,43 @@ Upgrade Pre-requisites
...
+    administrators need to remember and reconfigure these username/password
+    pairs.
+
+After upgrade is complete, a procedure for configuring the OIDC provider (Dex)
+user store will be provided in the next version.
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]: Maybe leave this in the warning block

**Review by gdemonet — CHANGES_REQUESTED** [review]

File: docs/operation/upgrade.rst (hunk 6 above)
Comment (Ebaneck) [reply docs/operation/upgrade.rst]: done

File: docs/operation/upgrade.rst (hunk 7 above)
Comment (Ebaneck) [reply docs/operation/upgrade.rst]: Done

File: docs/operation/upgrade.rst (hunk 8 above)
Comment (Ebaneck) [reply docs/operation/upgrade.rst]: Done

File: docs/operation/upgrade.rst [hunk 9]
```diff
@@ -24,14 +24,43 @@ Upgrade Pre-requisites
...
+authentication. Both K8s and Grafana will be configured to make use of the same
+OIDC provider.
+
+.. warning::
+
+    Before running an upgrade from 2.4.x to 2.5.0 or higher, MetalK8s
+    administrators **must** ensure all static users defined in
+    ``/etc/kubernetes/htpasswd`` can be recreated, and if any, all users that
+    were defined in Grafana.
+    The upgrade procedure will result in all admin credentialss being reset
```
Comment (gdemonet) [inline docs/operation/upgrade.rst]:
```suggestion
    The upgrade procedure will result in all admin credentials being reset
```

**Review by gdemonet — APPROVED** [review]

**Comment by Ebaneck** [comment]
//bypass_build_status
